Framework/Core/include/Framework/ASoA.h

```diff
@@ -1988,6 +1988,63 @@ std::tuple<typename Cs::type...> getRowData(arrow::Table* table, T rowIterator,
 {
   return std::make_tuple(getSingleRowData<T, Cs>(table, rowIterator, ci, ai, globalIndex)...);
 }
+
+template <typename R, typename T, typename C>
+R getColumnValue(const T& rowIterator)
+{
+  return static_cast<R>(static_cast<C>(rowIterator).get());
+}
+
+template <typename R, typename T>
+using ColumnGetterFunction = R (*)(const T&);
+
+template <typename R, typename T, typename C>
+ColumnGetterFunction<R, T> createGetterPtr(const std::string_view& columnLabel, bool& found)
+{
+  if (found) {
+    return nullptr;
+  }
+
+  if constexpr (std::is_arithmetic_v<typename C::type> && std::is_convertible_v<typename C::type, R>) {
+    if constexpr (o2::soa::is_dynamic_v<C>) {
+      if constexpr (o2::framework::pack_size(typename C::bindings_t{}) == o2::framework::pack_size(typename C::callable_t::args{})) {
+        if (std::string_view(&columnLabel[1]) == std::string_view(C::columnLabel()) || columnLabel == std::string_view(C::columnLabel())) {
+          found = true;
+          return &getColumnValue<R, T, C>;
+        }
+      }
+    } else if constexpr (o2::soa::is_persistent_v<C> && !o2::soa::is_index_column_v<C>) {
+      if (columnLabel == std::string_view(C::columnLabel())) {
+        found = true;
+        return &getColumnValue<R, T, C>;
+      }
+    }
+  }
+
+  return nullptr;
+}
+
+template <typename R, typename T, typename... Cs>
+ColumnGetterFunction<R, T> getColumnGetterByLabel(o2::framework::pack<Cs...>, const std::string_view& columnLabel)
+{
+  std::string_view labelView(columnLabel);
```

Comment:
Why do you actually need this?

Reply:
It would be more flexible to choose columns based on JSON file prepared for every trained neural network in CCDB, than hardcode columns and need to handle many networks to be retrained after future neural network design change. It is also possible to use this function to prepare a more common interface for NN inference for other ML projects in ALICE than ours.

I have updated the draft PR with the usage of this "getGetter" function in PIDML code, and you can take a look at file "pidONNXModel.h" in this PR https://github.com/AliceO2Group/O2Physics/pull/7371

Reply:
I did not explain myself. I mean the actual `labelView` copy you create here. Why can't you use `columnLabel` directly?

Reply:
Oh, I am sorry I misunderstood, my bad. You are right this copy is unnecessary.